game/zx_spectrum_tap.ksy

```diff
@@ -0,0 +1,82 @@
+meta:
+  id: zx_spectrum_tap
+  file-extension: tap
+  endian: le
+  license: CC0-1.0
+  title: ZX Spectrum tape file format
+  xref:
+    justsolve: TAP_(ZX_Spectrum)
+    pronom: fmt/801
+doc-ref: https://faqwiki.zxnet.co.uk/wiki/TAP_format
+seq:
+  - id: block
+    type: block
+    repeat: eos
+enums:
+  flag_enum:
+    0x00: header
+    0xFF: data
+  header_type_enum:
+    0: program
+    1: num_array
+    2: char_array
+    3: bytes
+types:
+  block:
+    seq:
+      - id: len_block
+        type: u2
+      - id: flag
+        type: u1
+        enum: flag_enum
+      - id: header
+        type: header
+        if: len_block == 0x13 and flag == flag_enum::header
+      - id: data
+        size: header.len_data + 4
+        if: len_block == 0x13
+      - id: headerless_data
+        size: len_block - 1
+        if: flag == flag_enum::data
+  header:
+    seq:
+      - id: header_type
+        type: u1
+        enum: header_type_enum
+      - id: filename
+        size: 10
+        pad-right: 0x20
+      - id: len_data
+        type: u2
+      - id: params
+        type:
+          switch-on: header_type
+          cases:
+            'header_type_enum::program': program_params
+            'header_type_enum::num_arry': array_params
+            'header_type_enum::char_arry': array_params
+            'header_type_enum::bytes': bytes_params
+      - id: checksum
+        type: u1
+        doc: Bitwise XOR of all bytes including the flag byte
+  program_params:
+    seq:
+      - id: autostart_line
+        type: u2
+      - id: len_program
+        type: u2
+  array_params:
+    seq:
+      - type: u1
+        doc: Unused
+      - id: var_name
+        type: u1
+        doc: Variable name (1..26 meaning A$..Z$ +192)
+      - type: u2
+        doc: Unused (0x8000)
```

Comment:
Should we specify this value as `contents`, since it's always supposed to be that value? On the other hand it's not an important part of the format, so maybe we shouldn't fail parsing if the value doesn't match.

Reply:
I guess it might make more sense to specify the expected value as it will help flag an issue with a corrupt file (or the parser grammar!). I think I'll add it. Sound OK?

Reply:
I really don't know what would be the best option here... You're right, it would help with detecting cases where something went wrong with the parsing. But I don't know how the Spectrum handles these unused fields when loading, or if they are even checked at all. If they aren't, there are *probably* tapes somewhere that have different values in those fields, but are otherwise valid. If that is the case, we should allow any value there.

It's always fun working with old formats where nobody really knows what's valid and how edge cases should be handled 😄

Reply:
I've just added the contents check for now.